mibidata/tiff.py

```diff
@@ -361,39 +371,59 @@ def _convert_from_previous(description):
         description['mibi.aperture'] = mi.MibiImage.parse_aperture(
             description['mibi.aperture'])
 
-def _get_page_data(page, description, metadata, channels):
+def _get_page_data(page, description, metadata, channels, inc_channels=None):
     """Adds to metadata and channel info for single TIFF page.
 
     Args:
         page: Single page in TIFF file.
         description: Decoded JSON description.
         metadata: Dictionary of metadata for entire TIFF file to add to.
         channels: List of channels for entire TIFF file to add to.
+        inc_channels: If set, only these channels will be added to channel
+                      parameter.
+
+    Returns:
+        include: Boolean representing whether channel should be included or not.
     """
-    channels.append((description['channel.mass'],
-                     description['channel.target']))
     # Get metadata on first SIMS page only
     if not metadata:
         metadata.update(_page_metadata(page, description))
+    if inc_channels:
+        if (description['channel.mass'],
+                description['channel.target']) not in inc_channels:
+            return False
+    channels.append((description['channel.mass'],
+                     description['channel.target']))
+    return True
 
-def info(filename):
+def info(filename, inc_channels=None):
     """Gets the metadata from a MibiTiff file.
 
     Args:
         filename: The path to the TIFF.
+        inc_channels: List or tuple of channel names to include in MibiImage.
+                      Names should be string tuples of format (mass, target).
 
     Returns:
         A dictionary of metadata as could be supplied as kwargs to
-        :class:`mibidata.mibi_image.MibiImage`, except with a ``channels`` key
-        whose value is a list of (mass, target) tuples.
+        :class:`mibidata.mibi_image.MibiImage`, except with a ``conjugates`` key
+        whose value is a list of (mass, target) tuples. If inc_channels is set,
+        then the metadata ``conjugates`` key only contains information for those
+        channels.
     """
+    if (inc_channels and not isinstance(inc_channels, (list, tuple)) and not
```

Comment:
Since I just reviewed DPPA-466, this code seems to be very similar to a couple other instances in that other PR. Once you merge DPPA-466 and bring the changes into here, would it make sense to refactor some of this code into helper functions like `check_is_list_of_tuples` or something?